base/missing.jl

```diff
@@ -107,16 +107,19 @@ max(::Missing, ::Any)     = missing
 max(::Any,     ::Missing) = missing
 
 # Rounding and related functions
-for f in (:(ceil), :(floor), :(round), :(trunc))
+round(::Missing, ::RoundingMode=RoundNearest; sigdigits::Integer=0, digits::Integer=0, base::Integer=0) = missing
+round(::Type{>:Missing}, ::Missing, ::RoundingMode=RoundNearest) = missing
+round(::Type{T}, ::Missing, ::RoundingMode=RoundNearest) where {T} =
+    throw(MissingException("cannot convert a missing value to type $T: use Union{$T, Missing} instead"))
+round(::Type{T}, x::Any, ::RoundingMode=RoundNearest) where {T>:Missing} = round(nonmissingtype(T), x)
+# to fix ambiguities
+round(::Type{T}, x::Rational, ::RoundingMode=RoundNearest) where {T>:Missing} = round(nonmissingtype(T), x)
+round(::Type{T}, x::Rational{Bool}, ::RoundingMode=RoundNearest) where {T>:Missing} = round(nonmissingtype(T), x)
+
+for f in (:(ceil), :(floor), :(trunc))
     @eval begin
-        ($f)(::Missing, digits::Integer=0, base::Integer=0) = missing
-        ($f)(::Type{>:Missing}, ::Missing) = missing
-        ($f)(::Type{T}, ::Missing) where {T} =
-            throw(MissingException("cannot convert a missing value to type $T: use Union{$T, Missing} instead"))
-        ($f)(::Type{T}, x::Any) where {T>:Missing} = $f(nonmissingtype(T), x)
-        # to fix ambiguities
-        ($f)(::Type{T}, x::Rational) where {T>:Missing} = $f(nonmissingtype(T), x)
-        ($f)(::Type{T}, x::Rational{Bool}) where {T>:Missing} = $f(nonmissingtype(T), x)
+        ($f)(::Missing; kwargs...) = round(missing; kwargs...)
+        ($f)(::Type{T}, ::Missing) where {T} = round(T, missing)
```

Comment:
Honestly it would be much clearer to list the supported keyword arguments here.

Reply:
I was emulating how things are defined in float.jl. The advantage of this, I suppose, is that we only need to specify the allowed keywords for `round()`, which also needs to support the `RoundingMode` argument, so can't be taken care of in a loop with the other functions.

Reply:
But the difference is that in float.jl there's an additional `::RoundingMode` argument. Here it's really confusing to call `round` from other functions.

Anyway if you define a method for `round`, better put it in the loop so that it's defined for all functions. You can define additional methods for `round` outside of the loop if needed.

Reply:
I have now returned to the original loop and just split out round() so the `::RoundingMode` argument can be taken care of.